daemons/controld/controld_fencing.c

```diff
@@ -845,7 +855,21 @@ te_fence_node(crm_graph_t *graph, crm_action_t *action)
  */
 
 static crm_trigger_t *stonith_history_sync_trigger = NULL;
-static mainloop_timer_t *stonith_history_sync_timer = NULL;
+static mainloop_timer_t *stonith_history_sync_timer_short = NULL;
+static mainloop_timer_t *stonith_history_sync_timer_long = NULL;
+
+static void
+tengine_stonith_history_synced(stonith_t *st, stonith_event_t *st_event)
+{
+    if(stonith_history_sync_timer_short) {
+        mainloop_timer_stop(stonith_history_sync_timer_short);
+    }
+    if(stonith_history_sync_timer_long) {
+        mainloop_timer_stop(stonith_history_sync_timer_long);
+    }
```

Comment:
You're not freeing and NULLing the timers here, so you can't just check whether they exist ... you have to either free and NULL them, or check whether they're running instead

Reply:
Scratch that, calling mainloop_timer_stop() on a stopped timer won't hurt. But we should free the timers in crmd_exit()